chalk-solve/src/solve.rs

```diff
@@ -57,100 +60,6 @@ impl<I: Interner> Solution<I> {
         }
     }
 
-    /// There are multiple candidate solutions, which may or may not agree on
```

Comment:
On the other hand (continued from where these are copied to), the fact that several of these functions are only used in the recursive solver makes me wonder if this is mostly okay anyways.